packages/node-sdk/src/client.ts

```diff
@@ -0,0 +1,607 @@
+        Array.isArray(event.evalRuleResults),
+      "event rule results must be an array",
+    );
+    ok(
+      event.evalMissingFields === undefined ||
+        Array.isArray(event.evalMissingFields),
+      "event missing fields must be an array",
+    );
+
+    return await this.post("flags/events", {
+      action: event.action,
+      flagKey: event.flagKey,
+      flagVersion: event.flagVersion,
+      evalContext: event.evalContext,
+      evalResult: event.evalResult,
+      evalRuleResults: event.evalRuleResults,
+      evalMissingFields: event.evalMissingFields,
+    });
+  }
+
+  private getFeatureFlagDefinitionCache() {
+    if (!this._shared.featureFlagDefinitionCache) {
+      this._shared.featureFlagDefinitionCache = cache<FlagDefinitions>(
+        this._shared.refetchInterval,
+        this._shared.staleWarningInterval,
+        this._shared.logger,
+        async () => {
+          const res = await this.get<FlagDefinitions>("flags");
+
+          if (!isObject(res) || !Array.isArray(res?.flags)) {
+            return undefined;
+          }
+
+          return res;
+        },
+      );
+    }
+
+    return this._shared.featureFlagDefinitionCache;
+  }
+
+  /**
+   * Sets the user that is used for feature flag evaluation.
+   *
+   * @param userId - The user ID to set.
+   * @param opts.attrs - The attributes of the user (optional).
+   *
+   * @returns A new client with the user set.
+   * @throws An error if the user ID is not a string or the options are invalid.
+   * @remarks
+   * If the user ID is the same as the current company, the attributes will be merged, and
+   * the new attributes will take precedence.
+   **/
+  public withUser(
+    userId: string,
+    opts?: { attributes?: Attributes },
+  ): BucketClient {
+    ok(
+      typeof userId === "string" && userId.length > 0,
+      "userId must be a string",
+    );
+    ok(opts === undefined || isObject(opts), "opts must be an object");
+    ok(
+      opts?.attributes === undefined || isObject(opts.attributes),
+      "attributes must be an object",
+    );
+
+    const client = new BucketClient(this);
+    if (userId !== this._user?.userId) {
+      client._user = { userId, attrs: opts?.attributes };
+    } else {
+      client._user = {
+        userId: this._user.userId,
+        attrs: { ...this._user.attrs, ...opts?.attributes },
+      };
+    }
+
+    return client;
+  }
+
+  /**
+   * Sets the company that is used for feature flag evaluation.
+   *
+   * @param companyId - The company ID to set.
+   * @param opts.attrs - The attributes of the user (optional).
+   *
+   * @returns A new client with the company set.
+   * @throws An error if the company ID is not a string or the options are invalid.
+   * @remarks
+   * If the company ID is the same as the current company, the attributes will be merged, and
+   * the new attributes will take precedence.
+   **/
+  public withCompany(
+    companyId: string,
+    opts?: { attributes?: Attributes },
```

Comment:
do you already have something in mind for the other options? If not, let's make this simpler and just let is take the `attributes?: Attributes` as a second parameter without the `opts`. If we really want to add more options later, that's still possible.